go/vt/vtgate/executor.go

```diff
@@ -985,6 +985,10 @@ func (e *Executor) getPlan(ctx context.Context, vcursor *vcursorImpl, sql string
 	}
 	ignoreMaxMemoryRows := sqlparser.IgnoreMaxMaxMemoryRowsDirective(stmt)
 	vcursor.SetIgnoreMaxMemoryRows(ignoreMaxMemoryRows)
+	consolidator := sqlparser.Consolidator(stmt)
+	if consolidator != querypb.ExecuteOptions_CONSOLIDATOR_UNSPECIFIED {
+		vcursor.SetConsolidator(consolidator)
+	}
```

Comment:
this needs to be always set irrespective of `ExecuteOptions_CONSOLIDATOR_UNSPECIFIED` otherwise the last value set on the consolidator will continue 
e.g.
Q1: select /*vt+ CONSOLIDATOR=ENABLE */ ...
Q2: select ...
Q3: select  ...

Q2 and Q3 will also continue to consolidate the result.

Reply:
Good catch, fixed 😬